astropy/table/table.py

```diff
@@ -1196,10 +1196,14 @@ def __getitem__(self, item):
             out._groups = groups.TableGroups(out, indices=self.groups._indices,
                                              keys=self.groups._keys)
             return out
-        elif ((isinstance(item, np.ndarray) and len(item) == 0) or
+        elif ((isinstance(item, np.ndarray) and item.size == 0) or
               (isinstance(item, (tuple, list)) and not item)):
             # If item is an empty array/list/tuple then return the table with no rows
             return self._new_from_slice([])
+        elif ((isinstance(item, np.ndarray) and item.size == 1) or
```

Comment:
This extra clause is not a good idea; instead, add to the `if isinstance(item, (int, np.integer))` line above, something like `or isinstance(item, np.ndarray) and item.shape == () and item.dtype.kind == 'i'`
(please check that these would in fact work, I'm just guessing.  Alternatively, everything that should be usable as an an index should have an `__index__` method, so you could replace the whole with 
```
try:
    index = item.__index__()
except (AttributeError, TypeError):
    pass
else:
    # do index stuff
```
But this may be more trouble than it is worth. Anyway, just run some checks and see what is sensible